### PR TITLE
Fix deprecations in common module (#22209)

### LINKED ISSUE #22209: Fix usage of deprecated API in common module
### Description

The common module uses some deprecated APIs as well as partially unnecessary reflection which can be avoided.

Currently reflection is used to call the protected method `liquibase.Liquibase#resetServices` . Instead of using reflection the method could be exposed as a public method by subclassing the `Liquibase` class, e.g. `KeycloakLiquibase`.

### PR DESCRIPTION
**Pull request by thomasdarimont**
- Use charset in `Encode` class
- Replace reflective call to protected `Liquibase#resetServices()` with call to exposed public method on a custom subclass `KeycloakLiquibase`
- Remove usage of deprecated AccessController class in Reflections
- Mark unused SetAccessibleProvilegedAction and UnsetAccessibleProvilegedAction as deprecated for removal in Keycloak 27

Fixes #22209

### CODE REVIEW

**Review by ahus1 — CHANGES_REQUESTED** [review]
Thank you for this PR. I agree the Liquibase parts can be improved by using less reflection. 

The removal of the classes in keycloak-common should be done by first deprecating them, and later removing them in a future issue. 

The change in `Encode.java` should not be a problem as we set `-Dfile.encoding=UTF-8` in the `JAVA_OPTS` in `kc.sh`, still it would be good to mention it in the release notes as it could lead to a different encoding of URL parameters.

**Comment by thomasdarimont** [comment]
@ahus1 thanks for taking a look! I readded the removed classes and marked them for removal in KC27. I also added a brief note to the release notes.

File: common/src/main/java/org/keycloak/common/util/reflections/Reflections.java [hunk 1]
```diff
@@ -988,7 +987,7 @@ public static boolean isPrimitive(Type type) {
      * @throws IllegalAccessException
      * @throws InstantiationException
      */
-    public static <T> T newInstance(final Class<T> fromClass) throws ClassNotFoundException, IllegalAccessException, InstantiationException {
+    public static <T> T newInstance(final Class<T> fromClass) throws ClassNotFoundException, IllegalAccessException, InstantiationException, InvocationTargetException, NoSuchMethodException {
```
Comment (ahus1) [inline common/src/main/java/org/keycloak/common/util/reflections/Reflections.java:990]: Nitpick: Changing the signature of the method is also a breaking change at least when people want to compile their code. AFAIK this method is not used, so I'd suggest to either keep it as is for now, deprecate it for KC27 and eventually remove it.

File: common/src/main/java/org/keycloak/common/util/reflections/Reflections.java [hunk 2]
```diff
@@ -1006,8 +1005,8 @@ public static <T> T newInstance(final Class<T> fromClass) throws ClassNotFoundEx
      * @throws IllegalAccessException
      * @throws InstantiationException
      */
-    public static <T> T newInstance(final Class<?> type, final String fullQualifiedName) throws ClassNotFoundException, IllegalAccessException, InstantiationException {
```
Comment (ahus1) [inline common/src/main/java/org/keycloak/common/util/reflections/Reflections.java:1009]: Nitpick - as above, I suggest deprecation and removal.

**Review by ahus1 — CHANGES_REQUESTED** [review]
Thank you for the changes. Please add the deprecated methods to `changes-26_0_0.adoc` so people will know that that they'll need to change something.

**Review by ahus1 — DISMISSED** [review]
Thank you for this PR. I pushed one more commit as the updated class in `common` still contained additional exceptions in its signature. 

Thinking again about the docs, also decided to move all the docs changes to the `upgrading` guide as only those upgrading might be interested in them. The release notes usually only contain the highlights.

**Review by ahus1 — APPROVED** [review]

**Pull request merged**